src/main/java/com/gregtechceu/gtceu/common/machine/electric/WorldAcceleratorMachine.java

```diff
@@ -0,0 +1,225 @@
+                    getLevel().getBlockState(randomPos).randomTick(this.getLevel().getServer().getLevel(this.getLevel().dimension()), randomPos,GTValues.RNG);
+                i++;
+            }
+            return;
+        }
+        // else handle block entity mode
+        Direction.stream().forEach(dir ->{
+            BlockEntity blockEntity = this.getLevel().getBlockEntity(this.getPos().relative(dir));
+            if(blockEntity != null && canAccelerate(blockEntity))
+                tickBlockEntity(blockEntity);
+        });
+    }
+
+    public boolean drainEnergy(long toDrain) {
+        long resultEnergy = energyContainer.getEnergyStored() - toDrain;
+        if (resultEnergy >= 0L && resultEnergy <= energyContainer.getEnergyCapacity()) {
+            energyContainer.removeEnergy(toDrain);
+            return true;
+        }
+        return false;
+    }
+
+    private <T extends BlockEntity> void tickBlockEntity(@NotNull T blockEntity){
+            BlockPos pos = blockEntity.getBlockPos();
+            BlockEntityTicker<T> blockEntityTicker =  this.getLevel().getBlockState(pos).getTicker(this.getLevel(),(BlockEntityType<T>) blockEntity.getType());
+            for (int i = 0; i < speed-1; i++) {
+                assert blockEntityTicker != null;
+                blockEntityTicker.tick(this.getLevel(), this.getPos(), blockEntity.getBlockState(), blockEntity);
+            }
+    }
+
+    private boolean canAccelerate(BlockEntity blockEntity){
+        if(blockEntity instanceof PipeBlockEntity || blockEntity instanceof IMachineBlockEntity) return false;
+
+        generateWorldAcceleratorBlacklist();
+
+        final Class<? extends BlockEntity> blockEntityClass = blockEntity.getClass();
+        if (blacklistCache.containsKey(blockEntityClass)) {
+            return blacklistCache.getBoolean(blockEntityClass);
+        }
+
+        for (Class<?> clazz : blacklistedClasses.values()) {
+            if (clazz.isAssignableFrom(blockEntityClass)) {
+                // Is a subclass, so it cannot be accelerated
+                blacklistCache.put(blockEntityClass, false);
+                return false;
+            }
+        }
+
+        blacklistCache.put(blockEntityClass, true);
+        return true;
+    }
+
+    @Override
+    public void onLoad() {
+        super.onLoad();
+        subscribeServerTick(this::update);
+    }
+
+    @Override
+    protected @NotNull InteractionResult onScrewdriverClick(Player playerIn, InteractionHand hand, Direction gridSide, BlockHitResult hitResult) {
+        if (!isRemote()) {
+            isRandomTickMode = !isRandomTickMode;
+            playerIn.sendSystemMessage(Component.translatable(isRandomTickMode?"gtceu.machine.world_accelerator.mode_entity":"gtceu.machine.world_accelerator.mode_tile"));
+            scheduleRenderUpdate();
+        }
+        return InteractionResult.CONSUME;
+    }
+
+    @Override
+    public boolean isWorkingEnabled() {
+        return this.isWorkingEnabled;
+    }
```

Comment:
use a lombok getter. `@Getter`

Reply:
also looking at it you shouldn't need to override the isWorkingEnabled or active fields.

Reply:
As far as I found active is only set with the recipe logic but since this machine doesn't have that I think manually setting it would be easier.

Reply:
you can just... set it yourself?

Reply:
also the active state still isn't saved anywhere.